src/widgets/splits/SplitContainer.cpp

```diff
@@ -222,8 +222,13 @@ void SplitContainer::addSplit(Split *split)
                          [this, split](HighlightState state) {
                              if (this->tab_ != nullptr)
                              {
-                                 this->tab_->updateHighlightState(
-                                     state, split->getChannelView());
+                                 if (this->tab_->updateHighlightState(state, split->getChannelView()))
+                                 {
+                                     if (auto *notebook = dynamic_cast<Notebook *>(this->parentWidget()))
```

Comment:
Add a comment here - I imagine it's something along the lines of
"The highlight state of this tab changed, force refresh the notebook to ensure the tabs visibilities are recalculated"